### PR TITLE
store conns in ets. thread poolname thru to acceptors, make avail to …

### PR DESCRIPTION
**Pull request by andymck**
Refactors the storing of connection data in the acceptor_pool from a state stored map to an ets cache.

This is to enable the connection list to be accessed, non blockingly from grpcbox_acceptor as part of handling an "acceptor_continue" callback.  At this point we pull the current connection count and if that exceeds the max_connection count as defined in the grpcbox config ( defined in the sys config ) the connection is rejected.

As part of this, the pool name has had to be treaded through from grpcbox to the acceptor pool so that it is known by the acceptor_pool process and then each individual acceptor in the pool.

### CODE REVIEW

File: src/acceptor_pool.erl [hunk 1]
```diff
@@ -346,19 +388,32 @@ init(Name, Mod, Args, {ok, {#{} = Flags, [#{} = Spec]}}) ->
...
     end;
-init(_, _, _, ignore) ->
+init(_, _, _, _, ignore) ->
     ignore;
-init(_, Mod, _, Other) ->
+init(_, _, Mod, _, Other) ->
     {stop, {bad_return, {Mod, init, Other}}}.
 
+-spec make_ets_conn_cache(atom()) -> atom().
+make_ets_conn_cache(PoolName) ->
```
Comment (Vagabond) [inline src/acceptor_pool.erl:406]: Why is this named/public, can we make it anonymous/protected, or does the pid structure not allow for this?

Comment (andymck) [reply src/acceptor_pool.erl:406]: I defaulted to public whilst working out the usage, will change to protected

File: src/acceptor_pool.erl [hunk 2]
```diff
@@ -77,6 +77,12 @@
 -export([format_status/2]).
 -export([terminate/2]).
 
+%% cached connections api
+-export([save_conn/5,
+         take_conn/2,
+         saved_conns/1,
+         num_conns/1]).
```
Comment (Vagabond) [inline src/acceptor_pool.erl:84]: only num_conns seems to be used outside of this module?

Comment (andymck) [reply src/acceptor_pool.erl]: fixed here: fc36e27aca1b789b222d5af9af757eebbdb948ae

**Comment by Vagabond** [comment]
It's strange that you track the connections but don't limit them here, but I suppose it lets the callback implement more options around connection accept/reject.

**Review by Vagabond — APPROVED** [review]

**Comment by andymck** [comment]
The reason I don't limit the connections in the pool is that it has no awareness of the config defined in sys config...all setup with config is performed/applied in grpcbox

**Review by jeffgrunewald — COMMENTED** [review]
that `list_to_atom/1` call makes me nervous but i understand we aren't able to use `list_to_existing_atom/1` in this context; why do we need a named table instead of just saving the ref of the table in the acceptor_pool state and passing that to any acceptors?

**Comment by jeffgrunewald** [comment]
i think i understand the problem; you need to be able to set the pool name from outside of the acceptor_pool lib and then query it and making a getter function here in acceptor pool for retrieving the table reference so you wouldn't need to pass in the table name would be a blocking call and cause it's own problems. i also see we've been setting the pool name in grpcbox with `list_to_atom/1` all along anyway

**Review by jeffgrunewald — APPROVED** [review]

**Review by vihu — APPROVED** [review]

**Pull request merged**